src/main/java/com/tradeshift/HelloResource.java

```diff
@@ -2,25 +2,59 @@
 
 import com.tradeshift.model.HelloResult;
 import com.tradeshift.model.MessageModel;
+import com.tradeshift.model.RecentResult;
+import com.tradeshfit.model.dao.MessageDAO;
+import com.tradeshfit.model.dao.MessagesDAO;
 import com.tradeshift.service.HelloService;
+import com.tradeshift.service.MessagesService;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Controller;
 
 import javax.ws.rs.*;
 import javax.ws.rs.core.MediaType;
 
-@Path("/")
+@Path("/messages")
 @Controller
 public class HelloResource {
-
     @Autowired
     private HelloService helloService;
 
-    @GET
-    @Path("hello")
+    @Autowired
+    private MessagesService messagesService;
+
+    public HelloResource(){
+
+    }
+    public HelloResource(HelloService helloService, MessagesService messagesService){
+        this.helloService = helloService;
+        this.messagesService = messagesService;
+    }
+
+    @POST
+    @Path("names/{name}")
     @Produces(MediaType.APPLICATION_JSON)
-    public HelloResult index(@QueryParam("name") String name){
-        HelloResult result = new HelloResult(new MessageModel(helloService.formatName(name)));
+    public HelloResult names(@PathParam("name") String name){
+        String message;
+        try{
+            message = helloService.formatName(name);
+            messagesService.insert(message);
+        }
+        catch (NullPointerException exp){
+            message = "Error: Name can not be empty";
```

Comment:
This type of response should be mapped into a `400 Bad Request` to show the client that the request is not valid. Overall catching NPE's are a bit controversial, since it might catch null pointers that you didn't intend; I would rather do 
```java
if (name == null) {
   throw new BadRequestException();
} else {
   // do the right stuff here without nulls
}
```